bin/objectbox_model_generator/lib/src/code_chunks.dart

```diff
@@ -43,4 +46,59 @@ class CodeChunks {
         const ${name}_OBXDefs = EntityDefinition<${name}>(_${name}_OBXModelGetter, _${name}_OBXReader, _${name}_OBXBuilder);
       """;
   }
+
+  static String _queryConditionBuilder(ModelEntity readEntity) {
+    final ret = <String>[];
+    for (var f in readEntity.properties) {
+
+      final name = f.name;
+
+      // see OBXPropertyType
+      String fieldType;
+      switch(f.type) {
+        case OBXPropertyType.Bool:
+          fieldType = "Boolean";
+          break;
+        case OBXPropertyType.String:
+          fieldType = "String";
+          break;
+        float:
+        case OBXPropertyType.Double:
+          fieldType = "Double";
+          break;
+        case OBXPropertyType.Float:
+          continue float;
+        integer:
+        case OBXPropertyType.Int:
+          fieldType = "Integer";
+          break;
+        case OBXPropertyType.Byte:
+          continue integer;
+        case OBXPropertyType.Short:
+          continue integer;
+        case OBXPropertyType.Char:
+          continue integer;
+        case OBXPropertyType.Long:
+          continue integer;
```

Comment:
Doesn't this work in dart? 
```dart

        case OBXPropertyType.Int:
        case OBXPropertyType.Long:
        case OBXPropertyType.Short:
        case OBXPropertyType.Byte:
        case OBXPropertyType.Char:
          fieldType = "Integer";
```

Reply:
Nope, no fall-through, try this:

```dart
// switch ...
  label meh:
  case a:
    return something();
  case b:
    continue a; // to simulate fall-thru
  case c:
     somethingElse();
     break;
// ...
```

I fixed a crap load of things, please try again.